### ISSUE TITLE
Git LFS ref merges are marked as 'ready to commit' but they arent merged

### ISSUE THREAD

**Issue by carbonated-niko** [ForkIssues/TrackerWin#517]
When merging LFS refs with conflicts, fork says the file is ready to commit, and doesnt give the option to select ours or theirs, but the file isnt merged.

```
version https://git-lfs.github.com/spec/v1
<<<<<<< HEAD
oid sha256:db51d6ae13e8b4016eac73b1cf1c48a6dda9fff72411d8e557cb3877d0e9ab9a
=======
oid sha256:98190567d0ef18bf2c79fc9847dfe5eabb6d617a5cad71b4d04a86698aae8878
>>>>>>> master-content
size 4252

```

**Comment by DanPristupov** [comment]
Thank you for reporting this.

**Comment by TanyaPristupova** [comment]
I can’t reproduce the problem.

Is that possible that you staged the file without resolving the conflict (unfortunately this is still possible https://github.com/ForkIssues/TrackerWin/issues/520)?

**Comment by carbonated-niko** [comment]
Hmm, I actually havent seen this in awhile, I'll keep an eye out for it. 

No, this was pre-staging. It occurred when I began a merge, and looked at the merge conflicts shown in my unstaged changes. Fork said that the file is ready to commit with no merges needed, and then id commit a file with an unresolved conflict

**Comment by carbonated-niko** [comment]
Hello Dan and Tanya - we ran into this issue again and I figured I'd make you a repository to test it on, I and my coworkers can reproduce the behavior 100% of the time. Basically fork reports conflicting dlls as being resolved when they shouldnt be. It doesnt appear to be an LFS issue as i also put a png into lfs and that shows the conflict properly. Give it a shot with this repo, (make sure you have LFS) but just merge mergeTest into master

https://github.com/carbonated-niko/MergeRepo/commits/master

**Comment by carbonated-niko** [comment]
I would like to clarify that this is different than 520 as I dont expect fork to prevent me from not handling a merge conflict, but I do expect fork to let me know that a conflict existed in the first place

**Comment by DanPristupov** [comment]
@carbonated-niko thank you very much for the repository. I see the behavior which I can't explain now. Surprisingly, this problem only occurs on Windows. This can be a bug in LFS or may be not. I will ask people in the GitLFS tracker.

**Comment by DanPristupov** [comment]
I'm investigating this problem right now and I don't see a way to fix this quickly. Git considers LFS pointers as a text diff and sometimes automatically merges them (incorrectly).

Please consider to mark files as **binary** in .gitattributes as a workaround. This will prevent automatic plain text merge:

**.gitattributes**
```
*.dll filter=lfs diff=lfs merge=lfs binary
*.png filter=lfs diff=lfs merge=lfs binary
```

**Comment by DanPristupov** [comment]
I'd like to add some more info.

This is a strange problem between LFS and git. As you probably know, instead of storing a binary file, LFS stores it as a pointer in a text format. Git doesn't know about LFS and in some cases conflicts in LFS files can be wrongly resolved (because for git they look like a text).

When conflicts occur, git tries to automatically merge them. In some cases a LFS pointer conflict looks resolvable and git incorrectly merges it and breaks the LFS file format. After that Fork can only see that the file had a conflict in the past and it's already resolved now.

Currently the only way to prevent automatic merge I see is to mark files as binary. This seems to be correct, but may be there's a better solution.

**.gitattributes**
```
*.dll filter=lfs diff=lfs merge=lfs binary
```

**Comment by carbonated-niko** [comment]
Thanks for giving it a look!

I actually attempted that before I sent it to you, as most people suggested that. I found that binary didnt make a difference. I just reconfirmed it with new updates to the repo, 'altTest' and 'altMergeTest' (bad names, I know). If you merge altmergetest into alttest youll see the same behavior despite the attributes being set to binary.

**Comment by DanPristupov** [comment]
This problem must be fixed in 1.60. Please try it in your environment.

In 1.60 we also improved internal merger and now it shows previews for LFS files and images. However it doesn't work well with the `binary` attribute in `.gitattributes`. In a case you used it, it must be reverted back to `-text`

<img width="600" alt="Screenshot 2021-03-19 at 18 14 33" src="https://user-images.githubusercontent.com/618115/111817993-f6394e00-88de-11eb-9003-31fc39d1b235.png">

Example of a correct `.gitattributes`:
```
*.dll filter=lfs diff=lfs merge=lfs -text
*.png filter=lfs diff=lfs merge=lfs -text
```
